### PR TITLE
Added WeeklyProfileParameter schema

### PR DESCRIPTION
**Pull request by s-simoncelli**

### CODE REVIEW

**Comment by jetuk** [comment]
This LGTM, but it needs `cargo fmt` running on it to pass the linting.

File: pywr_schema/src/parameters/profiles.rs [hunk 1]
```diff
@@ -62,3 +62,16 @@ impl UniformDrawdownProfileParameter {
         HashMap::new()
     }
 }
+
+#[derive(serde::Deserialize, serde::Serialize, Debug, Clone, PywrParameter)]
+pub struct WeeklyProfileParameter {
+    #[serde(flatten)]
+    pub meta: Option<ParameterMeta>,
+    pub values: Option<Vec<f64>>,
```
Comment (jetuk) [inline pywr_schema/src/parameters/profiles.rs:70]: For the `MonthlyProfileParameter` we use `Option<[f64; 12]>` to force a fixed size array. Should this be  `Option<[f64; 52]>` instead? I know Pywr accepts 52 or 53 values, but always truncates it to 52.

**Review by jetuk — CHANGES_REQUESTED** [review]

File: pywr_schema/src/parameters/profiles.rs (hunk 1 above)
Comment (jetuk) [reply pywr_schema/src/parameters/profiles.rs:70]: Also, we should add the following fields to permit validating schemas that refer to external data or a table for the values. 

```
    #[serde(flatten)]
    pub external: Option<ExternalDataRef>,
    #[serde(flatten)]
    pub table_ref: Option<TableDataRef>,
```

Comment (s-simoncelli) [reply pywr_schema/src/parameters/profiles.rs:70]: I implemented the parameter to handle 52 or 53 values. I'll change it to force 52

Comment (jetuk) [reply pywr_schema/src/parameters/profiles.rs:70]: I wonder whether that could be a feature in Pywr-next? Either 52 or 53 week profile?

**Comment by jetuk** [comment]
I've merged this with main. I think it is fine to leave it as a `Vec<f64>` here. We'll make the pywr-next schema more specific.

**Review by jetuk — APPROVED** [review]

**Pull request merged**